tests/test_qaoa.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :mod:`pennylane.qaoa` submodule.
+"""
+import pytest
+import networkx as nx
+import numpy as np
+import pennylane as qml
+from pennylane import qaoa
+
+
+#####################################################
+
+graph = nx.Graph()
+graph.add_nodes_from([0, 1, 2])
+graph.add_edges_from([(0, 1), (1, 2)])
+
+
+class TestMixerHamiltonians:
+    """Tests that the mixer Hamiltonians are being generated correctly"""
+
+    def test_x_mixer_output(self):
+        """Tests that the output of the Pauli-X mixer is correct"""
+
+        num_qubits = 4
```

Comment:
Probably good to run test for a few different number of qubits, to make it more robust. Not necessary if you think this will catch all possible errors

Reply:
I don't think this is necessary. The mixer itself is already very basic, and I'm confident that this will catch all errors 🙂